spring-boot-starter-rocketmq/src/main/java/org/apache/rocketmq/spring/starter/RocketMQAutoConfiguration.java

```diff
@@ -0,0 +1,190 @@
+    public RocketMQTemplate rocketMQTemplate(DefaultMQProducer mqProducer,
+        @Autowired(required = false)
+        @Qualifier("rocketMQMessageObjectMapper")
+            ObjectMapper objectMapper) {
+        RocketMQTemplate rocketMQTemplate = new RocketMQTemplate();
+        rocketMQTemplate.setProducer(mqProducer);
+        if (Objects.nonNull(objectMapper)) {
+            rocketMQTemplate.setObjectMapper(objectMapper);
+        }
+
+        return rocketMQTemplate;
+    }
+
+    @Configuration
+    @ConditionalOnClass(DefaultMQPushConsumer.class)
+    @EnableConfigurationProperties(RocketMQProperties.class)
+    @ConditionalOnProperty(prefix = "spring.rocketmq", value = "nameServer")
+    @Order
+    public static class ListenerContainerConfiguration implements ApplicationContextAware, InitializingBean {
+        private ConfigurableApplicationContext applicationContext;
+
+        private AtomicLong counter = new AtomicLong(0);
+
+        @Resource
+        private StandardEnvironment environment;
+
+        @Resource
+        private RocketMQProperties rocketMQProperties;
+
+        private ObjectMapper objectMapper;
+
+        public ListenerContainerConfiguration() {
+        }
+
+        @Autowired(required = false)
+        public ListenerContainerConfiguration(
+            @Qualifier("rocketMQMessageObjectMapper") ObjectMapper objectMapper) {
+            this.objectMapper = objectMapper;
+        }
+
+        @Override
+        public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+            this.applicationContext = (ConfigurableApplicationContext) applicationContext;
+        }
+
+        @Override
+        public void afterPropertiesSet() throws Exception {
+            Map<String, Object> beans = this.applicationContext.getBeansWithAnnotation(RocketMQMessageListener.class);
+
+            if (Objects.nonNull(beans)) {
+                beans.forEach(this::registerContainer);
+            }
+        }
+
+        private void registerContainer(String beanName, Object bean) {
+            Class<?> clazz = AopUtils.getTargetClass(bean);
+
+            if (!RocketMQListener.class.isAssignableFrom(bean.getClass())) {
+                throw new IllegalStateException(clazz + " is not instance of " + RocketMQListener.class.getName());
+            }
+
+            RocketMQListener rocketMQListener = (RocketMQListener) bean;
+            RocketMQMessageListener annotation = clazz.getAnnotation(RocketMQMessageListener.class);
+            BeanDefinitionBuilder beanBuilder = BeanDefinitionBuilder.rootBeanDefinition(DefaultRocketMQListenerContainer.class);
+            beanBuilder.addPropertyValue("nameServer", rocketMQProperties.getNameServer());
+            beanBuilder.addPropertyValue("topic", environment.resolvePlaceholders(annotation.topic()));
+
+            beanBuilder.addPropertyValue("consumerGroup", environment.resolvePlaceholders(annotation.consumerGroup()));
```

Comment:
So many hardcode strings, I suggest use a class to centralize these string constants.

Reply:
Fixed